docs/contributing/incremental_build.md

```diff
@@ -0,0 +1,135 @@
+
+3. **Build Tools:** Ensure the dependencies for building are installed and available, like `cmake` and `ninja`. These are installable through the `requirements/build.txt`, or can be installed by your package manager.
+
+    ```bash
+    uv pip install -r requirements/build.txt --torch-backend=auto
+    ```
+
+## Setting up the CMake Build Environment
+
+The incremental build process is managed through CMake. You can configure your build settings using a `CMakeUserPresets.json` file at the root of the vLLM repository.
+
+### Create `CMakeUserPresets.json`
+
+Here is an example configuration. Create a file named `CMakeUserPresets.json` in the root directory of your vLLM clone and paste the following content. Adjust paths (like `VLLM_PYTHON_EXECUTABLE` and `CMAKE_CUDA_COMPILER`) and settings as per your system configuration.
+
+```json
+{
+    "version": 6,
+    "cmakeMinimumRequired": {
+        "major": 3,
+        "minor": 20,
+        "patch": 0
+    },
+    "configurePresets": [
+        {
+            "name": "debug",
+            "generator": "Ninja",
+            "binaryDir": "${sourceDir}/cmake-build-debug",
+            "cacheVariables": {
+                "CMAKE_CUDA_COMPILER": "/usr/local/cuda/bin/nvcc", // Adjust if your nvcc is elsewhere
+                "CMAKE_C_COMPILER_LAUNCHER": "ccache",
+                "CMAKE_CXX_COMPILER_LAUNCHER": "ccache",
+                "CMAKE_CUDA_COMPILER_LAUNCHER": "ccache",
+                "CMAKE_BUILD_TYPE": "Debug",
+                "VLLM_PYTHON_EXECUTABLE": "/home/user/venvs/vllm/bin/python", // Adjust to your Python executable from the editable install's virtual environment
+                "CMAKE_INSTALL_PREFIX": "${sourceDir}",
+                "CMAKE_CUDA_FLAGS": "",
+                "NVCC_THREADS": "8", // Adjust based on your CPU cores
+                "CMAKE_JOB_POOL_COMPILE": "compile",
+                "CMAKE_JOB_POOLS": "compile=64" // Adjust based on your CPU cores
+            }
+        },
+        {
+            "name": "release",
+            "inherits": "debug",
+            "binaryDir": "${sourceDir}/cmake-build-release",
+            "cacheVariables": {
+                "CMAKE_BUILD_TYPE": "Release"
+            }
+        }
+    ],
+    "buildPresets": [
+        {
+            "name": "debug",
+            "configurePreset": "debug",
+            "jobs": 64 // Corresponds to CMAKE_JOB_POOLS
+        },
+        {
+            "name": "release",
+            "configurePreset": "release",
+            "jobs": 64 // Corresponds to CMAKE_JOB_POOLS
+        }
+    ]
+}
+```
+
+**Key configurations in `CMakeUserPresets.json`:**
```

Comment:
this is really non-trivial, lol

Reply:
I made a quick script, we can improve it over time